react-app/src/components/dashboard/sendForm.jsx

```diff
@@ -0,0 +1,175 @@
+import React, { Component } from "react";
+import Joi from "joi-browser";
+import { Form, Button, Search, Popup, Message, Icon } from "semantic-ui-react";
+import http from "../../services/httpService";
+
+class SendForm extends Component {
+  state = {
+    data: {
+      amount: "",
+      message: "",
+      recipientId: ""
+    },
+    search: {
+      isLoading: false,
+      results: [],
+      value: ""
+    },
+    balance: 0,
+    success: false,
+    errors: {}
+  };
+
+  schema = {
+    balance: Joi.number(),
+    amount: Joi.number()
+      .integer()
+      .min(1)
+      .max(Joi.ref("balance"))
+      .required()
+      .label("Amount"),
+    message: Joi.string().label("Message"),
+    recipientId: Joi.string()
+      .required()
+      .label("Recipient")
+  };
+
+  componentWillReceiveProps({ balance }) {
+    this.setState({ balance });
+  }
+
+  handleResultSelect = (e, { result }) => {
+    const state = { ...this.state };
+    state.search.value = result.title;
+    state.data.recipientId = result.id;
+
+    this.setState(state);
+  };
+
+  handleSearchChange = async (e, { value }) => {
+    const search = { ...this.state.search };
+    search.isLoading = true;
+    search.value = value;
+
+    this.setState({ search });
+
+    http.get(`http://localhost:3900/api/users?query=${value}`).then(res => {
```

Comment:
For later, create issue
1. Add a todo, replace hardcode url
2. value maybe vulnerable to security isssues.